libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionView.java

```diff
@@ -976,18 +976,33 @@ private void updateDataFromBannerText(@NonNull BannerText primaryBannerText, Ban
   }
 
   private void loadPrimary(BannerText primaryBannerText) {
-    stepPrimaryText.setMaxLines(2);
+    if (!ViewUtils.isLandscape(getContext())) {
```

Comment:
@Guardiola31337 - this change should fix the https://github.com/mapbox/mapbox-navigation-android/issues/1672

Since I don't have such long road info, can you help to verify?

Reply:
@JunDai 
I couldn't reproduce [#1672](https://github.com/mapbox/mapbox-navigation-android/issues/1672) with the master branch itself. I don't think it's an issue

Reply:
as I discussed with @Guardiola31337 today, he will double check #1672 

I just searched but couldn't find such road info at my side. But from code, I think #1672 might be still there. This line can make sure it's fixed.

Reply:
I think the thing we first need to confirm is whether the information should be displayed in 1 line only or it can be taken to 2 lines. As shown in #1672, is the requirement to make sure the road name was displayed properly in 2nd line or was the requirement to ellipsize the text in 1 line itself. Also, if there is no `secondaryText` then the `primary text` is allowed to be displayed in 2 lines as per existing logic in master branch. This makes sense to me, because since there is no secondary text then we can very well display the primary text in 2 lines. With the above logic change introduced in this PR, the existing logic will break.
cc @JunDai @Guardiola31337

Reply:
I would say the existing logic doesn't mean to show the primaryText in two lines if no secondaryText in landscape mode, because the landscape mode doesn't have enough space to show two lines. You could check the layout file.
* portrait mode, the primaryText and secondaryText are displayed in separate lines that's why we set primaryText's maxLines to 2 when secondaryText is absent.
* landscape mode, due to the limitation of the height of the screen, but in the other hand we get more space in width, we show primaryText and secondaryText in one line. 

If we want to show two lines, we need to either increase the height or decrease the font. Which we could do, but not good IMO.

Reply:
> as I discussed with @Guardiola31337 today, he will double check #1672 

Confirmed that I wasn't able to reproduce anymore 👀 https://github.com/mapbox/mapbox-navigation-android/issues/1672#issuecomment-640618473

Reply:
Yes exactly right. The current logic only handles portrait mode. But with your changes it will only handle the landscape mode and not the portrait mode since you check for `isLandscape` and only then assign maxLines=2, which in my opinion is not right because portrait mode needs 2 lines more than landscape. Since @Guardiola31337  was able to confirm that #1672 is not an issue anymore, let's revert this change which is trying to fix #1672. 
cc @JunDai

Reply:
@abhishek1508 
> Since @Guardiola31337  was able to confirm that #1672 is not an issue anymore, let's revert this change which is trying to fix #1672. 

Didn't OP aim to fix https://github.com/mapbox/mapbox-navigation-android/issues/1201? It's different from https://github.com/mapbox/mapbox-navigation-android/issues/1672

cc @JunDai

Reply:
@Guardiola31337 I meant only this particular change that is trying to fix #1672. Not the complete PR

Reply:
Ohh got it. Makes sense 👍

Reply:
> The current logic only handles portrait mode. 

Why the current logic only handles portrait mode? Isn't it set to 2 lines for both landscape and portrait mode?

> But with your changes it will only handle the landscape mode and not the portrait mode since you check for `isLandscape` and only then assign maxLines=2, which in my opinion is not right because portrait mode needs 2 lines more than landscape.

Can you check my change again? Isn't it setting the maxLines to 2 only for portrait?

I can revert this change if @Guardiola31337 can't reproduce this issue. But I would think there is risk that #1672 could happen on some devices as the code suggests.

cc @abhishek1508

Reply:
Let's go with it 👍